unitary/alpha/qudit_state_transform.py

```diff
@@ -0,0 +1,203 @@
+        result = result << 1
+    return result
+
+
+def qudit_to_qubit_state(
+    qudit_dimension: int,
+    num_qudits: int,
+    qudit_state_vector: np.ndarray,
+    pad_value: np.complex_ = 0,
+) -> np.ndarray:
+    """Converts a qudit-space quantum state vector to m-qubit-per-qudit column vector.
+
+    Each qudit is replaced by a set of qubits. Since the set of qubits can represent a larger state
+    space than the qudit, the state vector needs to be padded with 0s for those extra elements.
+    Based on https://drive.google.com/file/d/1n1Ym7JdnM44NnvNQDUXSk5rBLTQZzJ9t and
+    https://colab.research.google.com/drive/1MC6D4FyOXG0RjvzyV9IFcdRsLbUwwAcx
+
+    Args:
+        qudit_dimension: The dimension of a single qudit i.e. the number of states it can
+          represent.
+        num_qudits: The number of qudits in the given state vector.
+        qudit_state_vector: A flat, 1-D numpy array representing the state vector in qudit form.
+        pad_value: The value to set for the elements in the extra state space created in the
+          conversion. This field is mostly private, mostly for use by other methods. Do not set
+          this unless you really need to.
+
+    Returns:
+        A flat numpy array representing the input state vector using qubits.
+    """
+    # Reshape the state vector to a `num_qudits` rank tensor.
+    state_tensor = qudit_state_vector.reshape((qudit_dimension,) * num_qudits)
+    # Number of extra elements needed in each dimension if represented using qubits.
+    padding_amount = _nearest_power_of_two_ceiling(qudit_dimension) - qudit_dimension
+    # Expand the number of elements in each dimension by the padding_amount. Fill
+    # the new elements with the pad_value.
+    padded_state_tensor = np.pad(
+        state_tensor, pad_width=(0, padding_amount), constant_values=pad_value
+    )
+    # Return a flattened state vector view of the final tensor.
+    return np.ravel(padded_state_tensor)
+
+
+def qubit_to_qudit_state(
+    qudit_dimension: int,
+    num_qudits: int,
+    qubit_state_vector: np.ndarray,
+) -> np.ndarray:
+    """Converts a m-qubit-per-qudit column vector to a qudit-space quantum state vector.
+
+    Each qudit was replaced by a set of qubits. Since the set of qubits could represent a larger
+    state space than the qudit, the state vector needs to be sliced up to the qudit length in each
+    dimension.
+    Based on https://drive.google.com/file/d/1n1Ym7JdnM44NnvNQDUXSk5rBLTQZzJ9t and
+    https://colab.research.google.com/drive/1MC6D4FyOXG0RjvzyV9IFcdRsLbUwwAcx
+
+    Args:
+        qudit_dimension: The dimension of a single qudit i.e. the number of states it can
+          represent.
+        num_qudits: The number of qudits in the given/output state vector.
+        qubit_state_vector: A flat, 1-D numpy array representing the state vector in an
+          m-qubit-per-qudit form.
+
+    Returns:
+        A flat numpy array representing the input state vector using qudits.
```

Comment:
Same here.  Good to state dimensions for clarity's sake.

Reply:
Added